nedap/utils.modular
# Add `delegator`

## Brief


Helper for creating delegators. 

Delegators can be used to wrap existing impls with extra functionality such as dispatching events, authorization, logging etc.

## QA plan

* green tests
* use in its original project
* remove duplicate parts
* run linter, verify lack of `resource` warns



## Author checklist



    - [x] clj
    - [ ] cljs
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
      - [ ] append README

## Reviewer checklist

* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
